shared/lib/themes/actions.js

```diff
@@ -212,9 +241,21 @@ var ThemesActions = {
 	}
 };
 
-function hasSiteChanged() {
-	return ThemesLastQueryStore.hasSiteChanged() ||
-			! ThemesLastQueryStore.hasParams();
-}
+export function purchase( theme, site, source = 'unknown' ) {
+	const CartActions = require( 'lib/upgrades/actions' );
+	const themeItem = require( 'lib/cart-values/cart-items' ).themeItem;
+
+	return dispatch => {
+		CartActions.addItem( themeItem( theme.id, source ) );
 
-module.exports = ThemesActions;
+		defer( () => {
+			page( '/checkout/' + site.slug );
```

Comment:
`site.slug` is a computed attribute, which is not appropriate for the Redux state tree and will be removed in a future revision. See #2757 for tracking issue and 74052e7160bfc8d91f3d78fda4f2d1b4c919951b (#4055) for an example alternative.